rdt/hyper_transformer.py

```diff
@@ -367,30 +372,35 @@ def detect_initial_config(self, data):
                 Data which will have its configuration detected.
         """
         # Reset the state of the HyperTransformer
-        self.field_sdtypes = {}
-        self.field_transformers = {}
+        self.provided_field_sdtypes = {}
+        self._fitted_field_sdtypes = {}
+        self.provided_field_transformers = {}
+        self._fitted_field_transformers = {}
 
         # Set the sdtypes and transformers of all fields to their defaults
         for field in data:
             self._set_field_sdtype(data, field)
-            field_sdtype = self.field_sdtypes[field]
-            self.field_transformers[field] = get_default_transformer(field_sdtype)
+            field_sdtype = self._fitted_field_sdtypes[field]
+            self._fitted_field_transformers[field] = get_default_transformer(field_sdtype)
```

Comment:
I think all of this logic (except for resetting the provided dicts) can be moved into a `_learn_config` method that can also be called at the beginning of `fit`.